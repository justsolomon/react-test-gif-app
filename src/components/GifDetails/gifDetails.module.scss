@import "../../assets/scss/base/variables";

.gif {
  &-details {
    display: flex;
    justify-content: space-between;
  }

  &-container {
    width: 70%;

    img {
      max-width: 100%;
    }
  }

  &-title {
    font-weight: $font-weight--bold;
    margin-bottom: 0.7rem;
  }

  @media #{$media-tablet} {
    &-details {
      flex-direction: column-reverse;
    }

    &-container {
      width: 100%;

      img {
        width: 100%;
      }
    }
  }
}
