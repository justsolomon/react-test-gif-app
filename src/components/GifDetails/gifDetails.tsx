import { IGif } from "types/global";
import { UserDetails } from "./subcomponents/UserDetails";
import styles from "./gifDetails.module.scss";
import Skeleton from "react-loading-skeleton";

const GifDetails = ({ gif, loading }: { gif: IGif; loading: boolean }) => {
  return (
    <div className={styles["gif-details"]}>
      <UserDetails
        user={gif?.user}
        loading={loading}
        source={gif?.source || gif?.source_post_url}
      />
      <div className={styles["gif-container"]}>
        <p className={styles["gif-title"]}>
          {gif?.title || <Skeleton width={300} />}
        </p>
        {gif.images ? (
          <img src={gif?.images?.original?.url} alt="Original gif url" />
        ) : (
          <Skeleton height={300} width={500} />
        )}
      </div>
    </div>
  );
};

export default GifDetails;
