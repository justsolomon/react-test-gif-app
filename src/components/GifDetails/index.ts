export { default as GifDetails } from "./gifDetails";
