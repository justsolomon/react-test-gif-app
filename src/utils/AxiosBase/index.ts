import axios from "axios";

// Creates a base instance for all axios based request
const instance = axios.create({
  baseURL: `https://api.giphy.com/v1/`
});

instance.interceptors.request.use((config) => {
  //add api key to query params
  config.params["api_key"] = process.env.REACT_APP_API_KEY;

  return config;
});

export default instance;
