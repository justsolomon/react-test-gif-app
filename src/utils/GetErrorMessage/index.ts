import { AxiosError } from "axios";

const getErrorMessage = (err: AxiosError): string => {
  let errorMessage;
  if (err.response) errorMessage = err.response.data.message;
  else errorMessage = err.message;

  return errorMessage;
};

export default getErrorMessage;
