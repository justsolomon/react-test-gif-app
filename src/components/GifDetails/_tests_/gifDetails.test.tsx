import { render } from "@testing-library/react";
import { gif } from "utils/testUtils";
import GifDetails from "../gifDetails";

describe("<GifDetails />", () => {
  it("should render the GIF details component properly", () => {
    const { asFragment } = render(<GifDetails gif={gif} loading={false} />);

    expect(asFragment()).toMatchSnapshot();
  });
});
