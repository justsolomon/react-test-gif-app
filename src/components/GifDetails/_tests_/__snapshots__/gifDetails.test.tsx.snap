// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<GifDetails /> should render the GIF details component properly 1`] = `
<DocumentFragment>
  <div
    class="gif-details"
  >
    <div
      class="user-details"
    >
      <div
        class="user-details__group"
      >
        <img
          alt="User avatar"
          class="user-details__avatar"
          src="https://media3.giphy.com/avatars/roosterteeth/WSNaHIPXtMHL.jpg"
        />
        <div
          class="user-details__info"
        >
          <p
            class="user-details__display-name"
          >
            Rooster Teeth
          </p>
          <p
            class="user-details__username"
          >
            @roosterteeth
          </p>
        </div>
      </div>
      <p
        class="user-details__bio"
      >
        Premium dumb content. Home of Achievement Hunter, Funhaus, RWBY, RvB, a ton of podcasts, new stuff all the time, and the best community ever.
      </p>
      <button
        class="button  button--solid button--secondary "
        type="button"
      >
        View profile on Giphy
      </button>
      <div
        class="user-details__website"
      >
        <p
          class="user-details__website-header"
        >
          Source:
        </p>
        <a
          class="user-details__website-link"
          href="http://www.roosterteeth.com"
          rel="noreferrer"
          target="_blank"
        >
          <svg>
            link.svg
          </svg>
          <p
            class="user-details__website-link__text"
          >
            http://www.roosterteeth.com
          </p>
        </a>
      </div>
    </div>
    <div
      class="gif-container"
    >
      <p
        class="gif-title"
      >
        Work From Home Hello GIF by Rooster Teeth
      </p>
      <img
        alt="Original gif url"
        src="https://media3.giphy.com/media/QAU5Bd0Js9w3arZ4UD/giphy.gif"
      />
    </div>
  </div>
</DocumentFragment>
`;
